### ISSUE TITLE
Any plan to support protobuf 

### ISSUE THREAD

**Issue by nareshgnT** [springfox/springfox#3410]
- [ ] Feature Request. Any plan to support protobuf payloads in rest api using spring boot

**Comment by dilipkrish** [comment]
All the building blocks are available to do that. I can be implemented as a plugin

**Comment by nareshgnT** [comment]
@dilipkrish thanks for the response, let me know if I can help with

**Comment by dilipkrish** [comment]
Absolutely! if you'd like a high level overview I can set up some office hours to answer any questions. You could come up with a proposal and I'd be happy to accept your contribution.

At the outset this was my thinking
- Get familiar with how [ModelSpecificationProvider](https://github.com/springfox/springfox/blob/master/springfox-schema/src/main/java/springfox/documentation/schema/ModelSpecificationProvider.java) is implemented
- Get familiar with how [ModelPropertiesProvider](https://github.com/springfox/springfox/blob/master/springfox-schema/src/main/java/springfox/documentation/schema/property/ModelPropertiesProvider.java) is implemented
- Get familiar with how [ModelDependencyProvider.java](https://github.com/springfox/springfox/blob/master/springfox-schema/src/main/java/springfox/documentation/schema/ModelDependencyProvider.java) is implemented

With something like [Wire](https://github.com/square/wire/) you could parse the `.proto` files and implement the above to generate the specifications.
